Add board auto detection to the firmware flasher and add option to not automatically retrieve last selected board

Oats87 opened the pull request:
https://github.com/rotorflight/rotorflight-configurator/issues/110

This will need localization to the other supported languages in the configurator

rotorflight reviewed (changes requested):
Can you style the button a bit smaller?
It's like from another universe now.

Oats87 commented:
> Can you style the button a bit smaller? It's like from another universe now.

Updated the button positioning and added some additional logic to detect failure cases (and provide feedback)
